Homework03 - Flow Евтушенко Максим

evtushenko99 opened the pull request:
1. Поправил ошибку, метод LiveData.setValue можно вызывать только на главном потоке
2. Добавил подписку на stateFlow
3. Создал общий sealedClass Result,который помогает обрабатывать результат в активити
4. Добавил exceptionHandler для отображения ошибки

1. Реализовал методы task1-task4

Commit:
Homework03 - Flow Евтушенко Максим

1. Поправил ошибку, метод LiveData.setValue можно вызывать только на главном потоке
2. Добавил подписку на stateFlow
3. Создал общий sealedClass Result,который помогает обрабатывать результат в активити
4. Добавил exceptionHandler для отображения ошибки

1. Реализовал методы task1-task4

antonkazakov commented on flowcats/src/main/java/otus/homework/flowcats/presentation/CatsViewModel.kt:
```diff
@@ -0,0 +1,46 @@
+) : ViewModel() {
+
+    private val _catsLiveData = MutableLiveData<Fact>()
+    val catsLiveData: LiveData<Fact> = _catsLiveData
```
А зачем лайвдата тебе? Оставь один флоу типизированный Result и коллектор на lifecycleScope

evtushenko99 replied:
А я ее и не использую,  просто оставил для вида,что есть и такой вариант. Закомментировал строку,где ее использую.

evtushenko99 replied:
Удалил неиспользуемый код

antonkazakov commented on operators/src/main/java/otus/homework/flow/SampleInteractor.kt:
```diff
@@ -28,26 +38,44 @@ class SampleInteractor(
+    fun task2(): Flow<String> = sampleRepository.produceNumbers()
+        .transform { number: Int ->
+            emit(number.toString())
+            val resultString = StringBuilder("")
```
А зачем тебе стрингбилдер здесь? ты же можешь просто строки кидать, это оптимальнее решение

evtushenko99 replied:
Чтобы не делать дополнительные проверки на деление на 15. Так у меня две проверки и готова результирующая строка

evtushenko99 replied:
Я поправил решение, оно было не совсем корректное и тесты, но стрингбилдер оставил. Думаю это лучше, чем 3 проверки делать (может не прав)

antonkazakov reviewed (changes requested):
Привет. Пара замечаний

Commit:
Homework03 - Flow Евтушенко Максим часть2

1) Удалил неиспользуемый код в CatsViewModel
2) Поправил реализацию task2 в SampleInteractor

antonkazakov reviewed (commented):
👍
